src/main/capi/duckdb-c.cpp

```diff
@@ -8,7 +8,16 @@ using duckdb::DuckDB;
 duckdb_state duckdb_open_ext(const char *path, duckdb_database *out, duckdb_config config, char **error) {
 	auto wrapper = new DatabaseData();
 	try {
-		auto db_config = (DBConfig *)config;
+		DBConfig default_config;
+		DBConfig *db_config = &default_config;
+		DBConfig *user_config = (DBConfig *)config;
+		if (user_config) {
+			db_config = user_config;
+		}
+
+		if (db_config->options.duckdb_api.empty()) {
+			db_config->options.duckdb_api = "capi";
```

Comment:
Instead of doing the check here perhaps it's cleaner/easier to set `duckdb_api` to `"capi"` in `duckdb_create_config`?

Reply:
Some logic for setting `capi` is still needed here because `duckdb_open_ext()` is often called with `nullptr` passed in for config.